src/dsp_tools/utils/connection_live.py

```diff
@@ -298,6 +304,9 @@ def _handle_non_ok_responses(self, response: Response, request_url: str, retry_c
             msg = "Permanently unable to execute the network action. "
             if original_str := regex.search(r'{"knora-api:error":"dsp\.errors\.(.*)","@context', str(response.content)):
                 msg += f"\n{' '*37}Original Message: {original_str.group(1)}\n"
+                if original_str.group(1).startswith("OntologyConstraintException"):
+                    msg += f"See logs for more details: {logger_savepath}"
+                    raise InvalidInputError(msg)
```

Comment:
It's ok for this one, but generally I think we should work towards having structured information in representations of error state, not just "prose" strings describing the error.